UPGRADE-3.4.md

```diff
@@ -143,6 +143,37 @@ HttpKernel
            tags: ['console.command']
    ```
 
+ * The `getCacheDir()` method of your kernel should return the non-empty value
+   from `Kernel::getWarmupDir()` when one exists; not doing so is deprecated.
+
+   Before:
+
+   ```php
+   class AppKernel extends Kernel
+   {
+       public function getCacheDir()
+       {
+           return 'my-cache-dir';
+       }
+
+       // ...
+   }
+   ```
+
+   After:
+
+   ```php
+   class AppKernel extends Kernel
+   {
+       public function getCacheDir()
+       {
+           return $this->getWarmupDir() ?: 'my-cache-dir';
```

Comment:
I don't like the fact that an implementation detail leaks (`getWarmupDir()` is really an internal detail). I would prefer to not change this method at all. `getCacheDir()` is public because it's part of `HttpKernelInterface` to provide an easy way to override the cache dir in the project's kernel. It was never meant to be called directly, and most devs probably use `kernel.cache_dir` anyway.

Instead, we could add a phpdoc warning explaining what I've just written :)